packages/core/src/decorators/Transactional.spec.ts

```diff
@@ -0,0 +1,180 @@
+import { Test } from '@nestjs/testing';
+import { Inject, Injectable } from '@nestjs/common';
+import { setTimeout } from 'timers/promises';
+import {
+  getTransactionFromContext,
+  NoopTransaction,
+  NoopTransactionManagerModule,
+  PropagationLevel,
+  Transactional
+} from '../';
+
+describe('Transactional', () => {
+  it('should reuse existing tx in child', async () => {
+    // given
+    @Injectable()
+    class ChildService {
+      @Transactional({
+        propagation: PropagationLevel.Required,
+      })
+      public async testOperation() {
+        await setTimeout(1);
+        return getTransactionFromContext();
+      }
+    }
+
+    @Injectable()
+    class ParentService {
+      constructor(@Inject(ChildService) private childService: ChildService) {
+      }
+
+      @Transactional({
+        propagation: PropagationLevel.Required,
+      })
+      public async testOperation() {
+        await setTimeout(1);
```

Comment:
`await setTimeout(1);` bym wydzielił w testach do jakiejś ładnej funkcji bo jak to czytasz to w sumie nie wiesz po co ty te sleepy robisz?